charms/worker/k8s/src/charm.py

```diff
@@ -149,6 +149,26 @@ def __init__(self, *args):
             self.etcd = EtcdReactiveRequires(self)
             self.framework.observe(self.on.get_kubeconfig_action, self._get_external_kubeconfig)
 
+        # kubernetes-info
+        if self.lead_control_plane:
+            self.framework.observe(self.on.ceph_k8s_info_relation_changed, self._k8s_info)
```

Comment:
aight, so this doesn't really jive with the `reconcile` pattern presented in this charm.  by default, the control-plane charm will handle `ceph-k8s-info relation-changed` events through the `_reconcile(...)` method without doing this framework observe.  By registering this observe here: the charm is now handling with two methods:
`_reconcile` and `_k8s_info`

I'd recommend every control-plane unit (not just the leader) registers this event just as the reconciler will do
```suggestion
        if self.is_control_plane:
            self.framework.observe(self.on.ceph_k8s_info_relation_changed, self._k8s_info)
```

or maybe just do away with this entirely and put the `self._k8s_info` call into the `_reconcile(...)` method

Reply:
I suggest making the call to `self._k8s_info()` [here](https://github.com/canonical/k8s-operator/pull/119/files#diff-5ba1576e3daac57729756610a1f51b9706129e66f250d7a8557ea5666d0c490aR602)